### PR TITLE
Handle Bad Incoming Regex, Ensure that we throw the appropriate exception out of Admin/CreateInstance method

### PR DESCRIPTION
**Pull request by scbedd**

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Admin.cs [hunk 1]
```diff
@@ -162,7 +162,18 @@ private object GenerateInstance(string typePrefix, string name, HashSet<string>
                 }
             }
 
-            return Activator.CreateInstance(t, arg_list.ToArray());
+            try
+            {
+                return Activator.CreateInstance(t, arg_list.ToArray());
+            }
+            catch(Exception e)
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Admin.cs:169]: When activator.CreateInstance hit's an exception during its execution, it doesn't actually throw a `HttpException` directly.  It throws an InvocationException _wrapping_ the actual exception. We handle this here before rethrowing the appropriate exception. This code means that we don't need to do any wonky untangling of exceptions on caller side.

The new section in HttpExceptionMiddleware will handle generalized exceptions.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/AdminTests.cs [hunk 2]
```diff
@@ -395,6 +396,60 @@ public async void TestAddSanitizerThrowsOnBadRecordingId()
...
         }
 
+
+        [Fact]
+        public async Task GenerateInstanceThrowsOnBadBodyFormat()
+        {
+            RecordingHandler testRecordingHandler = new RecordingHandler(Directory.GetCurrentDirectory());
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Headers["x-abstraction-identifier"] = "UriRegexSanitizer";
+            httpContext.Request.Body = TestHelpers.GenerateStreamRequestBody("{\"value\":\"replacementValue\",\"regex\":[\"sv=2020-08-04&ss=bfqt&srt=sco&sp=rwdlacupitfx&se=2023-08-03T16:52:15Z&st=2021-08-03T08:52:15Z&spr=https&sig=randomsigM%3D\"]}");
```
Comment (sima-zhu) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/AdminTests.cs:406]: Though we currently do not have CredScan on tool repo, I know the CredScan tool will complain about this. Could you make it not secret like regex or read secret from file?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/StringSanitizer.cs [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
 
 namespace Azure.Sdk.Tools.TestProxy.Common
 {
     public static class StringSanitizer
     {
+        /// <summary>
+        /// Quick and easy abstraction for checking regex validity. Passing null explicitly will result in a True return.
+        /// </summary>
+        /// <param name="regex">A regular expression.</param>
+        /// <returns>True if the regex will successfully compile or if the regex is null. False otherwise.</returns>
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/StringSanitizer.cs:15]: Void method so you can probably delete this line.

Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/StringSanitizer.cs:15]: Refactor. 👎

Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/StringSanitizer.cs:15]:
```suggestion
```

**Review by benbp — APPROVED** [review]

**Pull request merged**